### PR TITLE
Replace traverse with neotraverse

### PR DESCRIPTION
**Pull request by wojtekmaj**
### Description

This PR replaces traverse with neotraverse, a fork and TypeScript rewrite of traverse with 0 dependencies (as opposed to 66: https://npmgraph.js.org/?q=traverse) and lots of improvements.

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [ ] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/plugins/validate-semantic/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as selectors from "./selectors"
 import * as actions from "./actions"
-import traverse from "traverse"
+import traverse from "neotraverse"
```
Comment (PuruVJ) [inline src/plugins/validate-semantic/index.js:3]: May or may not need to be turned into neotraverse/legacy, whether it works or not

**Comment by char0n** [comment]
Thanks a lot, 

SwaggerEditor@4 is technically EOL. It's been replaced  by SwaggerEditor@5 living on `next` branch. We're no longer want to make changes unless it's a security concern.
